### PR TITLE
chore: update link color tokens

### PR DESCRIPTION
**Pull request by dierat**
### Summary:
In https://github.com/chanzuckerberg/edu-design-system/pull/1280 I updated the link styles to use just 2 colors (neutral 800 and brand 700). This PR updates the tokens to match the new usage.

I spoke at length with design about which token names to use, and we settled on one token for each variant for color and border (4 tokens total). These are the only tokens designers outside of EDS will interact with because they don't use hover and focus states in their mocks. Therefore, I'm using the border tokens for the background focus states since we didn't add background tokens and they're all the same color anyway.

#### New tokens
- `eds-theme-color-text-link-neutral` -- #21272D -- `eds.color.neutral.800`
- `eds-theme-color-text-link-brand` -- #5751D2 -- `eds.color.brand.grape.700`
- `eds-theme-color-border-link-neutral` -- #21272D -- `eds.color.neutral.800`
- `eds-theme-color-border-link-brand` -- #5751D2 -- `eds.color.brand.grape.700`

#### Old tokens
- `eds-theme-color-text-link-default` -- #21272D -- `eds.color.neutral.800`
  - this can be replaced with `eds-theme-color-text-link-neutral` or `eds-theme-color-border-link-neutral` depending on whether it was styling text or border/background
- `eds-theme-color-text-link-default-hover` -- #161B1F -- `eds.color.neutral.black`
  - this wasn't being used
- `eds-theme-color-text-link-strong` -- #5751D2 -- `eds.color.brand.grape.700`
  - this can be replaced with `eds-theme-color-text-link-brand` or `eds-theme-color-border-link-brand` depending on whether it was styling text or border/background
- `eds-theme-color-text-link-strong-hover` -- #3E42B1 -- `eds.color.brand.grape.800`
  - this wasn't being used

### Test Plan:
- no chromatic changes (except for the token list)
- manually test links' default, hover, and focus states are unchanged

### CODE REVIEW

**Comment by dierat** [comment]
https://app.shortcut.com/czi-edu/story/214834/update-link-color-tokens-in-code

File: src/components/ClickableStyle/ClickableStyle.module.css [hunk 1]
```diff
@@ -14,7 +14,7 @@
   justify-content: center;
   align-items: center;
   /* Adds a gap between children to automatically handle spacing between icons and text. */
-  gap: 0.75rem; 
+  gap: 0.75rem;
```
Comment (dierat) [inline src/components/ClickableStyle/ClickableStyle.module.css:17]: These are just automated prettier updates.

**Review by ahuth — APPROVED** [review]
LGTM 🤘

**Comment by dierat** [comment]
Thanks for reviewing!

**Pull request merged**